pkg/ipam/node.go

```diff
@@ -357,7 +365,7 @@ func (n *Node) recalculate() {
 	n.available = a
 	n.stats.UsedIPs = len(n.resource.Status.IPAM.Used)
 	n.stats.AvailableIPs = len(n.available)
-	n.stats.NeededIPs = calculateNeededIPs(n.stats.AvailableIPs, n.stats.UsedIPs, n.getPreAllocate(), n.getMinAllocate(), n.getMaxAllocate())
+	n.stats.NeededIPs = calculateNeededIPs(n.stats.AvailableIPs, n.stats.UsedIPs, n.getPreAllocate(), n.getMinAllocate(), n.getMaxAllocate(scopedLog))
```

Comment:
Why passing the `scopedLog` here? You can just do `n.logger()` in `getMaxAllocate` method, right? There is nothing we decorate the logger with from what I can see.